frontend/src/components/Chat/SingleMessage.tsx

```diff
@@ -13,13 +14,16 @@ export default function SingleMessage({ message, myPlayerID }: SingleMessageProp
   const isSentFromUs = userId === myPlayerID;
   const className = isSentFromUs ? 'sent-from-us' : 'sent-from-them';
   const lastFourDigits = userId.slice(-4);
+  const userNameToRender = `${fromUserName}#${lastFourDigits}`;
 
   return (
     <div data-testid={className} className={`message-container ${className}`}>
       {isSentFromUs && <div data-testid='first-spacer' className='spacer first-spacer' />}
       <div className='message-content'>
         <div className='message-details'>
-          <div className='message-username'>{fromUserName}#{lastFourDigits}</div>
+          <div className='message-username'>
+            <Tooltip label={userNameToRender}>{userNameToRender}</Tooltip>

```

Comment:
Curious why you changed to tooltip for the username?

Reply:
I added some css that accounts for names that are too long to fit in their container—if they're too long, they'll truncate with an ellipsis. But we still want people to be able to read the full names if they really want to, so I added a tooltip that lets them see full names on hover

Reply:
Gotcha!